ocs_ci/ocs/md_blow.py

```diff
@@ -0,0 +1,149 @@
+import logging
+import threading
+from time import sleep
+
+from ocs_ci.ocs.resources.pod import get_noobaa_core_pod, get_noobaa_db_pod
+from ocs_ci.ocs.resources.mcg import MCG
+
+logger = logging.getLogger(__name__)
+
+
+class MdBlow(object):
+    """
+    Upload objects directly into DB using MD Blow script present in core pod
+    """
+
+    def __init__(self):
+        """
+        Assign default values to its parameter and patch required lines in core pod
+        """
+        self.db_percentage = None
+        self.obj_count = 1000
+        self.concurrency = 50
+        self.chunks = 1
+        self.chunk_size = 200
+        mcg_obj = MCG()
+        self.creds = mcg_obj.get_noobaa_admin_credentials_from_secret()
+        self.email = self.creds["email"]
+        self.password = self.creds["password"]
+        self.noobaa_core_pod = get_noobaa_core_pod()
+        self.noobaa_db_pod = get_noobaa_db_pod()
+
+    def patch_lines_in_core_pod(self):
+        """
+        patch required lines in script to avoid failures
+        seed below line in js file at line no 32
+        await require('../server/system_services/master_key_manager.js').get_instance().load_root_keys_from_mount();
+        """
+        f_path = "/root/node_modules/noobaa-core/src/tools/md_blow.js"
+        insert_at = 31
+        line = (
+            "        await require("
+            '"../server/system_services/master_key_manager.js"'
+            ").get_instance().load_root_keys_from_mount();"
+        )
+        cmd = f"sed -i '{insert_at}a\\{line}' {f_path}"
+        self.noobaa_core_pod.exec_cmd_on_pod(cmd)
+        logger.info(f"Patched {line} line in md_blow.js file")
+
+    def monitor_db_usage(self, db_percentage):
+        """
+        Get current DB percentage
+
+        Args:
+            db_percentage (int): Percentage of DB to be monitored
+        """
+        current_db_percentage = self.noobaa_db_pod.exec_cmd_on_pod(
+            "df -h | grep postgresql | awk '{print $5}'",
+            shell=True,
+        )
+        current_db_usage = int(current_db_percentage.strip().replace("%", ""))
+        logger.info(current_db_usage)
+        if current_db_usage >= db_percentage:
+            logger.warn(
+                f"Provided fill percentage is invalid. Given {db_percentage}... Current usage of DB {current_db_usage}"
+            )
+            self.stop_dumping.set()
+        if db_percentage >= 100:
```

Comment:
You could simply assert `assert db_percentage < 100` in the beginning if the value is 100 or more?

Reply:
How about we move these validations to `upload_obj_using_md_blow`?

If `db_percentage` is lower than the current, we return since the target has been reached.
If it's invalid (not a valid percentage number) we can raise an exception before we start any threads.